File: frontend/src/types/category.ts
export interface Category {
  id: number;
  company_id: number;
  name: string;
  description: string;
  status: "Active" | "Inactive";
  product_count?: number;
  created_at: string;
  updated_at: string;
}

export interface CategoryFormData {
  name: string;
  description: string;
  status: "Active" | "Inactive";
}

File: backend/app/schemas/category.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field


class CategoryBase(BaseModel):
    name: str = Field(..., min_length=2, max_length=120)
    description: str | None = None
    status: str = "Active"


class CategoryCreate(CategoryBase):
    pass


class CategoryUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=120)
    description: str | None = None
    status: str | None = None


class CategoryResponse(CategoryBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    company_id: int
    product_count: int = 0
    created_at: datetime
    updated_at: datetime